/*
 * This file was automatically generated by EvoSuite
 * Wed Nov 27 04:19:55 GMT 2024
 */


import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true) 
public class SortHeap_ESTest extends SortHeap_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      Integer[] integerArray0 = new Integer[9];
      Integer integer0 = new Integer((-201));
      integerArray0[0] = integer0;
      integerArray0[1] = integer0;
      Integer integer1 = new Integer((-1));
      integerArray0[2] = integer1;
      integerArray0[3] = integerArray0[2];
      int int0 = 7;
      integerArray0[4] = integerArray0[0];
      integerArray0[5] = integer0;
      integerArray0[6] = integer0;
      Integer[] integerArray1 = SortHeap.sort(integerArray0, (Integer) int0);
      assertEquals(9, integerArray1.length);
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      Integer[] integerArray0 = new Integer[9];
      Integer integer0 = new Integer(4174);
      integerArray0[1] = integer0;
      // Undeclared exception!
      try { 
        SortHeap.sort(integerArray0, integerArray0[1]);
        fail("Expecting exception: ArrayIndexOutOfBoundsException");
      
      } catch(ArrayIndexOutOfBoundsException e) {
         //
         // no message in exception (getMessage() returned null)
         //
      }
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      SortHeap sortHeap0 = new SortHeap();
  }
}
